lib/src/git.rs

```diff
@@ -221,15 +221,18 @@ pub fn import_some_refs(
         // Apply the change that happened in git since last time we imported refs
         let full_name = to_git_ref_name(ref_name).unwrap();
         mut_repo.set_git_ref_target(&full_name, new_git_target.clone());
-        mut_repo.merge_single_ref(ref_name, old_git_target, new_git_target);
-        // If a git remote-tracking branch changed, apply the change to the local branch
-        // as well
-        if !git_settings.auto_local_branch {
-            continue;
-        }
-        if let RefName::RemoteBranch { branch, remote: _ } = ref_name {
-            let local_ref_name = RefName::LocalBranch(branch.clone());
-            mut_repo.merge_single_ref(&local_ref_name, old_git_target, new_git_target);
+        if let RefName::RemoteBranch { branch, remote } = ref_name {
+            // Remote-tracking branch is the last known state of the branch in the remote.
+            // It shouldn't diverge even if we had inconsistent view.
```

Comment:
I haven't looked at this carefully, but wouldn't the remote branches occasionally diverge anyway because of concurrent operations? Or perhaps we should somehow pick a winner in that case as well?

I'm worried that if diverged remote branches happen, but only very rarely, we won't support the special case of diverged remote branches well enough.

Reply:
Good point about the concurrent operation. In that case, both `git_refs` and `remote_branches` could diverge when merging views, and will be "fixed" on the next `import_refs()`. Before this commit, `remote_branches` would be left diverged whereas `git_refs` knows the true remote tracking branches (stored in the Git repository.)

Does that make sense?

Reply:
Yes, good point, this sounds great to me.

Could you maybe insert a comment explaining this?

It would also benefit from a test (not necessarily in this commit). I actually have a draft test that doesn't quite work, so if you don't feel like it and I don't forget, I will probably eventually write one.

Reply:
> Before this commit, `remote_branches` would be left diverged

I was wrong. `import_refs()` can resolve conflicts because both `remote_branches` and old `git_refs` should have the same conflicting refs (`old_conflict - old_conflict + new = new`). So for the "concurrent operation" scenario, there would be no behavior change.

Maybe I can add some tests for that, but I'm not gonna include them in this PR.

Reply:
I'm not sure what you mean by 

> for the "concurrent operation" scenario, there would be no behavior change.

and your remark about `import_refs`.

Do you mean that remote branches that got conflicted because of concurrency will stay conflicted? In what situation can `import_refs` resolve conflicts?

If the answer to the first question is "yes", I'm a bit worried. Judging by your tone, though, I'm guessing it's a "no" on that question.

Reply:
> Do you mean that remote branches that got conflicted because of concurrency will stay conflicted?

No.

Without this PR, `merge_single_ref(remote_branch, ...)` resolves the conflict to `new_git_target` (because `old_git_target` matches.) With this PR, `set_remote_branch_target()` overwrites the conflicting target to `new_git_target`. So the end result is the same.

Reply:
I guess you mean that this happens even if `old_git_target` is conflicted.

In any case, I can think about it later and/or we can write tests.

Reply:
> I guess you mean that this happens even if `old_git_target` is conflicted.

Yes, and that's the situation we're discussing. 

After "concurrent operations", two views are merged, and both `git_refs["refs/remotes/{remote}/{name}"]` and `branches[name].remote_targets[remote]` get conflicted in the same way. Applying diff of `git_refs[..]` on `remote_targets[..]` succeeds because the old `git_refs[..] == remote_targets[..]`.